pkg/chunk/gcp/table_client.go

```diff
@@ -0,0 +1,86 @@
+package gcp
+
+import (
+	"strings"
+
+	"cloud.google.com/go/bigtable"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/status"
+
+	"github.com/weaveworks/cortex/pkg/chunk"
+)
+
+type tableClient struct {
+	cfg    Config
+	client *bigtable.AdminClient
+}
+
+// NewTableClient returns a new TableClient.
+func NewTableClient(ctx context.Context, cfg Config) (chunk.TableClient, error) {
```

Comment:
I found a blog about this recently!  I thought of you.  The reason they gave is that it makes it easier to find the implementation.  Meh.

The struct is private anyway, so exposing it would give a lint error.